rx_signal_processing/rx_dsp_chain.cu

```diff
@@ -15,406 +15,234 @@
-    if (converges < 0){ // REVIEW #10 remez returns True or False is this check correct?
-        std::cerr << "Filter failed to converge with cutoff of " << filter_cutoff // REVIEW #5 & #34 print units
-            << ", transition width " << transition_width << ", and rate "
-            << rate << std::endl;
-        //TODO(keith): throw error
+    //Then receive packet from driver
+    zmq::message_t driver_request;
+    driver_socket.recv(&driver_request);
+    rxsamplesmetadata::RxSamplesMetadata rx_metadata;
+    std::string driver_str(static_cast<char*>(driver_request.data()), driver_request.size());
+    if (rx_metadata.ParseFromString(driver_str) == false) {
+      //TODO(keith): handle error
     }
 
-    std::vector<std::complex<float>> complex_taps;
-    for (auto &i : filter_taps) {
-        auto complex_tap = std::complex<float>(i,0.0);
-        complex_taps.push_back(complex_tap); // REVIEW #0 does this work?
-                                             // REPLY This is just converting real numbers to complex numbers with 0 in the imag part.
-                                              //  I dont think this needs to be here anymore. It can be done during the mixing.
+    DEBUG_MSG("Got driver request");
+
+    //Verify driver and radar control packets align
+    if (sp_packet.sequence_num() != rx_metadata.sequence_num()) {
+      //TODO(keith): handle error
+      DEBUG_MSG("SEQUENCE NUMBER mismatch radar_control: " << sp_packet.sequence_num()
+        << " usrp_driver: " << rx_metadata.sequence_num());
     }
 
-    //TODO(keith): either pad to pwr of 2 or make pwr of 2 filter len
-    //Pads to at least len 32 or next pwr of 2 to stop seg fault for now.
-
-    //Quick and small lambda to get the next power of 2. Returns the same
-    //number if already power of two.
-    auto next_pwr2 = [](uint32_t n) -> uint32_t {
-        n--;
-        n |= n >> 1;
-        n |= n >> 2;
-        n |= n >> 4;
-        n |= n >> 8;
-        n |= n >> 16;
-        n++;
-        return n;
-
-    };
-
-    uint32_t size_difference = 0;
-    if (complex_taps.size() < 32) {
-        size_difference = 32 - complex_taps.size();
+    //Parse needed packet values now
+    if (sp_packet.rxchannel_size() == 0) {
+      //TODO(keith): handle error
     }
-    else {
-        size_difference = next_pwr2(complex_taps.size()) - complex_taps.size();
+    std::vector<double> rx_freqs;
+    for(int i=0; i<sp_packet.rxchannel_size(); i++) {
+      rx_freqs.push_back(sp_packet.rxchannel(i).rxfreq());
     }
-    //pad the filter with the result.
-    for (uint32_t i=0 ; i<size_difference; i++) {
-        complex_taps.push_back(std::complex<float>(0.0,0.0));
+
+    auto mix_timing_start = std::chrono::steady_clock::now();
+
+    filters.mix_first_stage_to_bandpass(rx_freqs,rx_rate);
```

Comment:
Hmm... you're continually adding to the vector with push_back (in mix_first_stage_to_bandpass) but it doesn't look like you ever clear it?